core/src/main/java/nl/nn/adapterframework/doc/DocWriterNew.java

```diff
@@ -0,0 +1,470 @@
+import static nl.nn.adapterframework.doc.DocWriterNewXmlUtils.addSequence;
+import static nl.nn.adapterframework.doc.DocWriterNewXmlUtils.getXmlSchema;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.Logger;
+
+import nl.nn.adapterframework.doc.model.ConfigChild;
+import nl.nn.adapterframework.doc.model.ConfigChildKey;
+import nl.nn.adapterframework.doc.model.ElementType;
+import nl.nn.adapterframework.doc.model.FrankAttribute;
+import nl.nn.adapterframework.doc.model.FrankDocModel;
+import nl.nn.adapterframework.doc.model.FrankElement;
+import nl.nn.adapterframework.util.LogUtil;
+import nl.nn.adapterframework.util.XmlBuilder;
+
+public class DocWriterNew {
+	private static final String CONFIGURATION = "nl.nn.adapterframework.configuration.Configuration";
+
+	private static Logger log = LogUtil.getLogger(DocWriterNew.class);
+
+	private FrankDocModel model;
+	private Map<String, String> syntax2Names;
+	List<SortKeyForXsd> xsdSortOrder;
+	private XmlBuilder xsdRoot;
+
+	public DocWriterNew(FrankDocModel model) {
+		this.model = model;
+		Map<String, List<FrankElement>> simpleNamePartitions = model.getAllElements().values().stream()
+				.collect(Collectors.groupingBy(
+						FrankElement::getSimpleName,
+						Collectors.toList()));
+		syntax2Names = new HashMap<>();
+		for(List<FrankElement> partition: simpleNamePartitions.values()) {
+			syntax2Names.putAll(chooseSyntax2Names(partition));
+		}
+	}
+
+	public void init() {
+		xsdSortOrder = breadthFirstSort(CONFIGURATION);
+	}
+
+	static Map<String, String> chooseSyntax2Names(List<FrankElement> elementPartition) {
+		Map<String, String> result = new HashMap<>();
+		if(elementPartition.size() == 1) {
+			FrankElement theElement = elementPartition.get(0);
+			result.put(theElement.getFullName(), theElement.getSimpleName());
+			return result;
+		}
+		else {
+			List<List<String>> nameComponents = new ArrayList<>();
+			for(FrankElement element: elementPartition) {
+				List<String> packageNameComponents = Arrays.asList(element.getFullName().split("\\.", -1));
+				if(! element.getSimpleName().equals(packageNameComponents.get(packageNameComponents.size() - 1))) {
+					log.warn(String.format("Syntax 2 names may be wrong because there is a FrankElement with full name [%s] but simple name [%s]",
+							element.getFullName(), element.getSimpleName()));
+				}
+				nameComponents.add(packageNameComponents);
+			}
+			List<String> fullNames = elementPartition.stream().map(elem -> elem.getFullName()).collect(Collectors.toList());
+			int numComponents = 2;
+			while(true) {
```

Comment:
Kan je in twee regels commentaar uitleggen wat hieronder gebeurt?

Reply:
Dit was functionaliteit die we bij nader inzien niet willen. Deze code is weg.